src/kubernetes_cluster/proof/controller_runtime_liveness.rs

```diff
@@ -169,4 +173,98 @@ pub proof fn lemma_true_leads_to_reconcile_scheduled_by_assumption<T>(reconciler
     schedule_controller_reconcile().wf1(cr_key, spec, next_and_cr_exists, pre, post);
 }
 
+pub proof fn lemma_reconcile_idle_leads_to_reconcile_idle_and_scheduled_by_assumption<T>(reconciler: Reconciler<T>, cr_key: ObjectRef)
+    requires
+        cr_key.kind.is_CustomResourceKind(),
+    ensures
+        sm_partial_spec(reconciler).and(always(lift_state(|s: State<T>| s.resource_key_exists(cr_key)))).entails(
+            lift_state(|s: State<T>| { !s.reconcile_state_contains(cr_key)})
+                .leads_to(lift_state(|s: State<T>| {
+                    &&& !s.reconcile_state_contains(cr_key)
+                    &&& s.reconcile_scheduled_for(cr_key)})
+            ),
+        )
+{
```

Comment:
Would it be easier if you use `schedule_controller_reconcile().wf1` here? Similar to the proof of `lemma_true_leads_to_reconcile_scheduled_by_assumption`. In that way, you don't need to introduce `leads_to_confluence_self_temp` I guess?

Reply:
I have trouble doing this. What I have is `|= pre /\ !post /\ next -> pre`, but what `wf1` requires is `|= pre /\ next -> pre || post`.